### ISSUE TITLE
raspiblitz.conf permissions

### ISSUE THREAD

**Issue by rek79** [raspiblitz/raspiblitz#2433]
The raspiblitz.conf config file is created under /mnt/hdd with permissions root:root -wr-r--r--r.  When installing bonus features such as sphinxrelay there are permission denied messages since the script is running under admin.   For instance, Line 158 of bonus.sphinxrelay.sh does not use sudo to make edits to this config file.  It seems like ownership of raspiblitz.conf should be changed to admin:admin so that sudo is not required.  Has anyone else encountered this?  Thoughts?

https://github.com/rootzoll/raspiblitz/blob/fc1d44104bc22d85ebebd7b0d6ee2240722a9edf/home.admin/40addHDD.sh#L107-L109
https://github.com/rootzoll/raspiblitz/blob/fc1d44104bc22d85ebebd7b0d6ee2240722a9edf/home.admin/config.scripts/blitz.migration.sh#L95-L97

![image](https://user-images.githubusercontent.com/9994060/126424267-7e8a879d-fa5a-4dd5-b357-27d0f9d808ad.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/bonus.sphinxrelay.sh#L156-L159

![image](https://user-images.githubusercontent.com/9994060/126424816-993f721e-a079-4084-98b4-cf60e23a5d89.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/bonus.mempool.sh#L66-L69

![image](https://user-images.githubusercontent.com/9994060/126424940-97d3ec60-46c5-4cc5-8d53-5438c24562d5.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/bonus.btc-rpc-explorer.sh#L67-L70)

![image](https://user-images.githubusercontent.com/9994060/126425104-445728d0-36c7-4ef1-bf55-d8985841ffa2.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/bonus.electrs.sh#L238-L241

![image](https://user-images.githubusercontent.com/9994060/126425279-0d8342d8-4c34-40ad-ba16-b3a8f398771f.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/bonus.thunderhub.sh#L55-L58

![image](https://user-images.githubusercontent.com/9994060/126425371-70c684e9-777d-42e3-b52e-8d34290d50f7.png)

https://github.com/rootzoll/raspiblitz/blob/7ff2ca32f29eb7383ae810dd0d3ffc688863b69c/home.admin/config.scripts/lnd.keysend.sh#L20-L22

**Comment by rek79** [comment]
If this is acknowledged as an issue, and changing ownership to admin is not feasible, then this may be a potential solution using "tee" command:

Source:
https://www.cyberciti.biz/faq/sudo-append-data-text-to-file-on-linux-unix-macos/

Method 1: Use tee command
The tee command read from standard input (such as keyboard) and write to standard output (such as screen) and files. The syntax is:
```
echo 'text' | sudo tee -a /path/to/file
echo '192.168.1.254 router' | sudo tee -a /etc/hosts
```

Sample outputs:

Password:
`192.168.1.254   router`
This solution is simple and you avoided running bash/sh shell with root privileges. Only append or write part needed root permission.

**Comment by rek79** [comment]
Confirmed working fix:
![image](https://user-images.githubusercontent.com/9994060/126527419-94ca072d-aefa-4c48-9c0d-d715eed71fa6.png)
![image](https://user-images.githubusercontent.com/9994060/126527485-cbcb8253-9232-4d72-bf37-8427e5cbca87.png)
`echo "BTCRPCexplorer=off" | sudo tee -a /mnt/hdd/raspiblitz.conf`

Should I prepare a PR to replace "`>>`" operators with " `| sudo tee -a`"? Or, optionally prepare a PR to change ownership of raspiblitz.conf to admin:admin?

**Comment by openoms** [comment]
Are you now building with the branch `pre-1.7.1` ? Because this clearly was not a problem in v1.7.0.

**Comment by rek79** [comment]
No.  I'm still using v1.7.  It could easily go unnoticed in v1.7 since the script is just establishing defaults in the config file.  I believe it would not stop the bonus features from actually functioning.  If you are running 1.7 and try typing `echo "BTCRCexplorer=off" >> /mnt/hdd/raspiblitz.conf` as admin you should get the same result.  This command would require logging in as root to be successful since the file is owned by root.

**Comment by openoms** [comment]
on my vanilla v1.7 node:
```
admin@David:~ $ ls -la /mnt/hdd/raspiblitz.conf 
-rwxrwxrwx 1 root root 367 Jul 21 22:01 /mnt/hdd/raspiblitz.conf
```

the `/mnt/hdd/raspiblitz.conf` is owned by root, but writable by all users.
The problem you encounter is that on your system the `raspiblitz.conf` has stricter permissions.

**Comment by rek79** [comment]
I should have noticed that. The config file must have been created by a different branch of code when I had trouble with localIP issue on Odroid HC2.  I'll review the code to see where this could have happened and report back.  We may need to check permissions after config creation if it doesn't happen normally at 40addHDD.sh

**Comment by rek79** [comment]
I looked through the code.  Every location that creates raspiblitz.conf seems to set 777 permissions right after.  I manually changed the permissions using chmod 777.  I'm not sure how they got altered.  I will most likely save blockchain data, wipe and start with a fresh image.  The most likely culprit is user error on my part while tinkering.  I'm going to close this issue.

**Comment by openoms** [comment]
when a script is run with `sudo` it runs with root user
so a command like:
`echo "anything" >> raspiblitz.conf`
would create the file with default permissions (not 777) if it wouldn't exit before.
It was probably a wrong order of scripts with some failing.

**Comment by rek79** [comment]
That makes a lot of sense.  I did manually run individual scripts with sudo to troubleshoot.  It was a valuable lesson learned, thanks.

**Comment by rootzoll** [comment]
The bootstrap script now makes sure that the `raspiblitz.conf` gets the chmod 664 with owner=root & group=sudo (admin user is in group sudo) this way every config-script called by root or admin can simply edit/write to the `raspiblitz.conf` but other users like additional apps can just read and not write to it. I think thats a good balance of security for now.

**Comment by rootzoll** [comment]
This will be part of the v1.7.1rc1. To be tested if:
- [x] after fresh setup raspiblitz.conf if 664 with root:sudo
- [x] after recover/update raspiblitz.conf if 664 with root:sudo
- [x] after migration raspiblitz.conf if 664 with root:sudo
- [ ] other scripts running fine with the new permission

**Comment by openoms** [comment]
@rootzoll in af3db7f I added the `cl.check.sh` which makes sure there are no config entries in the CL config which would stop it running. 
It runs with the `bitcoin` user which has no sudo access, but can uninstall the problematic plugins.

Can we add 777 permissions on raspiblitz.conf or can you think of a better solution to be able to switch those plugins off in the config if they are deactivated by `cl.check.sh`?

**Comment by rootzoll** [comment]
I would like to keep the `raspiblitz.conf` not writable to other users - especially so that other additional apps can just read it, but not write it. We can think about the `bitcoin` user to  the `sudo` group ... but the whole idea of having a seperate `bitcoin` user was to keep it seperated from the `admin` user with sudo rights - but that would be at least better than just setting `777` permissions on the `raspiblitz.conf` file.

**Comment by rootzoll** [comment]
Maybe having a seperate file for the cl plugins that can then be part of the CL data dir makes more sense anyway for the CL plugins? That way it would keep a clean seperation.

**Comment by rootzoll** [comment]
basic topic runs for v1.7.1 release ... moving to v1.8.0 to maybe discuss further the edge case of the `cl.check.sh` script.
